src/Orleans.Core.Abstractions/Streams/Core/StreamSequenceToken.cs

```diff
@@ -10,6 +10,16 @@ namespace Orleans.Streams
     [Serializable]
     public abstract class StreamSequenceToken : IEquatable<StreamSequenceToken>, IComparable<StreamSequenceToken>
     {
+        /// <summary>
+        /// Number of event batches in stream prior to this event batch
+        /// </summary>
+        public long SequenceNumber { get; protected set; }
```

Comment:
what does this mean to other stream providers which doesn't support rewindability

Reply:
It shouldn't matter for those, as they shouldn't rely on tokens.  In those cases, this information is only used to order data in the cache for delivery.

Reply:
The cache ordering requirement already existed even for non-rewindable stream providers, thus this is not a new constraint, it was just not expressed in the token.